### PR TITLE
[microTVM][Zephyr][projectAPI] Minimize project build commands

### PR DESCRIPTION
**Pull request by mehrdadh**
This PR changes Zephyr template project to populate all configs in Cmake file and minimize the command to build the project.

Before this PR we had to do this:
```
cmake -DARG1 -DARG2 ..
cd build
make -j2
```

After this PR:
```
cmake ..
cd build
make -j2
```

This PR also removes some of the redundancies in passing project options like zephyr_board.

cc @alanmacd @gromero @guberti

### CODE REVIEW

File: apps/microtvm/zephyr/template_project/microtvm_api_server.py [hunk 1]
```diff
@@ -419,7 +419,7 @@ def _create_prj_conf(self, project_dir, options):
             f.write("\n")
 
     API_SERVER_CRT_LIBS_TOKEN = "<API_SERVER_CRT_LIBS>"
-    ENABLE_CMSIS_TOKEN = "<ENABLE_CMSIS>"
+    CMAKE_ARGS = "<CMAKE_ARGS>"
```
Comment (gromero) [inline apps/microtvm/zephyr/template_project/microtvm_api_server.py:424]: I'd like to use  `CMAKE_ARGS_TOKEN` as the var name here , with `_TOKEN` at the end, like it was before for the `ENABLE_CMSIS_TOKEN` var.

File: gallery/how_to/work_with_microtvm/micro_tvmc.sh [hunk 2]
```diff
@@ -121,7 +121,9 @@ tvmc compile magic_wand.tflite \
 #
 # To generate a Zephyr project we use TVM Micro subcommand ``create``. We pass the MLF format and the path
 # for the project to ``create`` subcommand along with project options. Project options for each
-# platform (Zephyr/Arduino) are defined in their Project API server file. To generate Zephyr project, run:
+# platform (Zephyr/Arduino) are defined in their Project API server file. To build
+# Zephyr project for a different Zephyr board, change ``zephyr_board`` project option.
```
Comment (gromero) [inline gallery/how_to/work_with_microtvm/micro_tvmc.sh:125]: Please remove the trailing space in this line.

**Review by gromero — CHANGES_REQUESTED** [review]

File: apps/microtvm/zephyr/template_project/microtvm_api_server.py [hunk 3]
```diff
@@ -576,22 +578,33 @@ def build(self, options):
...
+    def _find_board_from_cmake_file(cls) -> str:
+        with open(API_SERVER_DIR / CMAKELIST_FILENAME) as cmake_f:
+            for line in cmake_f:
+                if line.startswith("set(BOARD"):
+                    zephyr_board = line.strip("\n").strip("set(BOARD ").strip(")")
+                    break
+
+        if not zephyr_board:
+            raise RuntimeError(
+                f"Zephyr Board is not found in the {API_SERVER_DIR / CMAKELIST_FILENAME}"
```
Comment (gromero) [inline apps/microtvm/zephyr/template_project/microtvm_api_server.py:599]: Could this message be changed to `"No Zephyr board defined in the {API_SERVER_DIR / CMAKELIST_FILENAME}"` ? Or `"set in the ...`" ?

File: apps/microtvm/zephyr/template_project/microtvm_api_server.py [hunk 4]
```diff
@@ -576,22 +578,33 @@ def build(self, options):
...
 
+    @classmethod
+    def _find_board_from_cmake_file(cls) -> str:
+        with open(API_SERVER_DIR / CMAKELIST_FILENAME) as cmake_f:
+            for line in cmake_f:
+                if line.startswith("set(BOARD"):
+                    zephyr_board = line.strip("\n").strip("set(BOARD ").strip(")")
+                    break
+
+        if not zephyr_board:
```
Comment (gromero) [inline apps/microtvm/zephyr/template_project/microtvm_api_server.py:597]: if no line starting with `set(BOARD`  is found here `zephyr_board` will be left undefined so `if not zephyr_board` will fail with `ErrorName`: var is not defined in `if` in line 597. So `zephyr_board = None` should be set just before `with` statement?

**Review by gromero — CHANGES_REQUESTED** [review]

**Comment by gromero** [comment]
Otherwise, LGTM :+1:

File: apps/microtvm/zephyr/template_project/microtvm_api_server.py (hunk 1 above)
Comment (mehrdadh) [reply apps/microtvm/zephyr/template_project/microtvm_api_server.py:424]: done

File: apps/microtvm/zephyr/template_project/microtvm_api_server.py (hunk 4 above)
Comment (mehrdadh) [reply apps/microtvm/zephyr/template_project/microtvm_api_server.py:597]: great catch!

File: apps/microtvm/zephyr/template_project/microtvm_api_server.py (hunk 3 above)
Comment (mehrdadh) [reply apps/microtvm/zephyr/template_project/microtvm_api_server.py:599]: done

**Review by mehrdadh — COMMENTED** [review]
@gromero thanks for the review! PTAL

**Review by gromero — APPROVED** [review]
@mehrdadh Thanks for addressing the comments. LGTM!

I'll just let the CI finish and give some time if somebody else wants to review it too.

File: tests/micro/common/test_tvmc.py [hunk 5]
```diff
@@ -53,7 +53,10 @@ def test_tvmc_exist(platform, board):
 @tvm.testing.requires_micro
 @pytest.mark.parametrize(
     "output_dir,",
-    [pathlib.Path("./tvmc_relative_path_test"), pathlib.Path(tempfile.mkdtemp())],
+    [
+        pathlib.Path("./tvmc_relative_path_test"),
+        # pathlib.Path(tempfile.mkdtemp())
```
Comment (gromero) [inline tests/micro/common/test_tvmc.py:58]: Is that line really to be a comment?

**Review by gromero — CHANGES_REQUESTED** [review]
Since there are new changes since my last approval, I'm doing another review cycle. PTAL.

File: tests/micro/common/test_tvmc.py (hunk 5 above)
Comment (mehrdadh) [reply tests/micro/common/test_tvmc.py]: thanks for catching, fixed it

**Review by gromero — APPROVED** [review]
LGTM. I'll just let the CI finish and give some time if somebody else wants to review it too.

**Pull request merged**